### PR TITLE
Modify TechDocsCollator and Search to be aware of new urls

### PR DESCRIPTION
**Pull request by Xantier**
## Hey, I just made a Pull Request!

* Add boolean constructor param to define if to use legacy paths
* Swap urls to match new pattern
* Add few tests
* Modify tech docs search initial filter to be aware of new url patterns

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by dtuite** [comment]
@techdocs-core The check "Automatically add new TechDocs Issues and PRs to the GitHub project board" seems to not be working.

**Comment by dtuite** [comment]
Regarding what this PR is about...

There is a recent Tech docs URLs change which changes things to to be all lowercase instead of what it is currently where the `kind`  is uppercased. 

The PR addresses that breaking change on two fronts: 
 1. when we are indexing the search data for tech docs
 2. when we are filtering the search on the tech docs page.

File: packages/backend/src/plugins/search.ts
Comment (iamEAP) [inline packages/backend/src/plugins/search.ts:76]: Good opportunity to create a static `DefaultTechDocsCollator::fromConfig()` equivalent?  I think catalog recently updated to do the same.

File: plugins/techdocs-backend/src/search/DefaultTechDocsCollator.ts
Comment (iamEAP) [inline plugins/techdocs-backend/src/search/DefaultTechDocsCollator.ts:87]: I don't know how common it is to have cased namespaces or entity names, but it'd be good to keep the logic in sync w/the rest of TechDocs (lowercase all the types).

File: .changeset/fluffy-peas-care.md
Comment (iamEAP) [inline .changeset/fluffy-peas-care.md:2]: Need to add `@backstage/plugin-techdocs` too, eh?

**Review by iamEAP — APPROVED** [review]
Ah-hah!  I skimmed the title of this and thought, "But isn't this taken care of because any case is handled now that all cases are handled?"  But that extra context in your comment was super helpful @dtuite!

Couple of minor comments (most important is the changeset), the other stuff is more nice-to-have.

File: .changeset/fluffy-peas-care.md
Comment (Xantier) [reply .changeset/fluffy-peas-care.md:2]: Correct :+1:

File: packages/backend/src/plugins/search.ts
Comment (Xantier) [reply packages/backend/src/plugins/search.ts:76]: :+1: I'll add a fromConfig static method and a deprecation flag to the constructor

**Comment by Xantier** [comment]
Thanks @iamEAP, I've addressed those comments now.

**Review by iamEAP — APPROVED** [review]
Nice!

**Comment by dtuite** [comment]
TechDocs approved! 🥳  Can we get a +1 from the core team pretty please?? 🙏

File: packages/backend/src/plugins/search.ts
Comment (freben) [inline packages/backend/src/plugins/search.ts:81]: Does this mean that we now have a config parameter in place, that may not actually be read by most installations so it can lead to confusion when people ask for support? Why is not this one read inside `fromConfig`? I don't get that bit.

Comment (Xantier) [reply packages/backend/src/plugins/search.ts]: Good point. I modified this to be read directly within the collator instead.

**Comment by Xantier** [comment]
This is now ready for another look from the core team to get search actually working again for tech docs.

File: packages/backend/src/plugins/search.ts
Comment (benjdlambert) [inline packages/backend/src/plugins/search.ts:76]: Does the previous way still work or should we be updating `create-app` and creating a migration note saying how to update?

File: plugins/techdocs-backend/src/search/DefaultTechDocsCollator.ts
Comment (freben) [inline plugins/techdocs-backend/src/search/DefaultTechDocsCollator.ts:176]: always `toLocaleLowerCase('en-US')` to be sure that you don't run into e.g. this https://en.wikipedia.org/wiki/Dotted_and_dotless_I

File: plugins/techdocs/src/reader/components/TechDocsSearch.tsx
Comment (freben) [inline plugins/techdocs/src/reader/components/TechDocsSearch.tsx:64]: same here

File: plugins/techdocs-backend/src/search/DefaultTechDocsCollator.ts
Comment (freben) [inline plugins/techdocs-backend/src/search/DefaultTechDocsCollator.ts:34]: this should probably actually be exported all the way out to the API so it appears in the api-report and makes the `ae-forgotten-export` go away. We know there's lots of those but we're trying to whittle them away. :)

**Review by freben — APPROVED** [review]
OK I think this looks great. Super minor things and then good to go

File: packages/backend/src/plugins/search.ts
Comment (Xantier) [reply packages/backend/src/plugins/search.ts:76]: The old way still works, it uses the constructor directly. There is a new optional parameter in the newer version of the constructor, `legacyPathCasing`, which defaults to false. For now it is fine, since TechDocsCollator/Search is not yet officially 'released' and is missing from create-app altogether.

**Review by Xantier — COMMENTED** [review]
Thanks @freben @benjdlambert, I've addressed those items now as well.

**Pull request merged**
